### PR TITLE
fixed uncertainy formatter and added tests

### PR DESCRIPTION
**Pull request by grburgess**
Fixes the issues in #451

### CODE REVIEW

**Review by omodei — COMMENTED** [review]
I can see that the test has passed successfully, can you also paste the output (as a comment)?

**Comment by grburgess** [comment]
@omodei ah, the only difference is that now inf will be reported as inf rather than crashing the code., e.g., 
```1 -.5 + inf``` 

and nans will report zero as they always have... however, the user will be warned when either of these occur.

File: threeML/test/test_basic.py [hunk 1]
```diff
@@ -288,3 +288,27 @@ def test_gbm_workflow():
     dl_files = p.glob("*_bkg.h5")
 
     [x.unlink() for x in dl_files]
+
+
+
+
+def test_uncertainty_formatter():
```
Comment (henrikef) [inline threeML/test/test_basic.py:295]: I guess we could explicitly test the return values here, but that might be overkill.

**Review by henrikef — APPROVED** [review]
Looks reasonable.

File: threeML/test/test_basic.py (hunk 1 above)
Comment (grburgess) [reply threeML/test/test_basic.py:295]: good point @henrikef I will add that and then merge

**Pull request merged**